app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/MainScreen.kt

```diff
@@ -40,11 +43,42 @@ fun MainScreen() {
         derivedStateOf { navBackStackEntry?.toMainDestination() ?: MainNavigation.startDestination }
     }
 
+    var lastStartDestination: MainNavigation by remember { mutableStateOf(MainNavigation.startDestination) }
```

Comment:
Use directly a callback from the screen, and each destination can set its own popbackstack